enhancements/security/security-profiles-operator.md

```diff
@@ -0,0 +1,629 @@
+
+## Summary
+
+Let's integrate the [Security Profiles Operator](https://github.com/kubernetes-sigs/security-profiles-operator)
+as part of the OpenShift platform in order to allow folks to:
+
+* Easily install security profiles to secure their workloads
+  (both SELinux and Seccomp)
+
+* Manage the lifecycle of the aforementioned profiles (make sure
+  that they get attached to the appropriate container as upgrades
+  happen)
+
+* Enable users to automatically generate profiles for their workloads
+
+* Provide predefined profiles for existing default workloads in OpenShift
+
+
+## Motivation
+
+OpenShift already has several security controls enabled by default. Amongst them
+are the following:
+
+* SELinux to restrict what resources can containers interact with
+
+* Seccomp to restrict the system calls that containers are allowed to do
+
+* Security Context Constraints to control what security knobs and handles are
+  service accounts able to tweak in their workloads.
+
+The default SELinux and Seccomp profiles work great for restricted workloads.
+However, once the workload needs a little more permissions, setting these
+security settings becomes non-trivial. In face of this, developers often
+respond by turning off these controls and running their workloads as privileged
+containers.
+
+The [Security Profiles Operator (SPO)](https://github.com/kubernetes-sigs/security-profiles-operator)
+is an upstream project aiming to ease installation and development of security
+profiles in Kubernetes.
+
+### Goals
+
+* Integrate the Security Profiles Operator into OpenShift by having it
+  installed by default via the Cluster Version Operator (CVO).
+
+* Integrate the SPO into OpenShift's SCCs.
+
+* Enable application developers to ship their operators with
+  relevant security profiles (OLM integration)
+
+### Non-Goals
+
+What is out of scope for this proposal? Listing non-goals helps to focus discussion
+and make progress.
+
+## Proposal
+
+The Security Profiles Operator should be installed by default in OpenShift
+via the CVO as any other core component. As it is, its main function is to
+install security policies and ensure that they're attached to the appropriate
+pods which require them.
+
+The installation is managed by a DaemonSet workload (Security Profiles Operator Daemon
+or SPOD) that listens for security profiles (either `SeccompProfile` or
+`SelinuxProfile`) and ensures that these are installed on the nodes.
+
+The SPOD instance is configurable through a CRD which allows us to set where
+it's scheduled and the capabilities that are enabled/disabled. For OpenShift,
+SELinux would be enabled by default and the DaemonSet would be scheduled
+everywhere.
+
+`SeccompProfiles` are handled by simply persisting a file on a directory that
+the Kubelet has access to. A `SeccompProfile` looks as follows:
+
+```yaml
+---
+apiVersion: security-profiles-operator.x-k8s.io/v1alpha1
```

Comment:
we don't want to ship alpha APIs, do we?

Reply:
we don't indeed

Reply:
All our API's are alpha for now (planned for the upcoming v0.3.0 release). We may consider graduating them and changing the release versioning model (otherwise the graduation would not make that much sense together with semver.

Created an issue about this topic in https://github.com/kubernetes-sigs/security-profiles-operator/issues/407